### ISSUE TITLE
Call to a member function id() on null when merging commits from the git integration in

### ISSUE THREAD

**Issue by robdekort** [statamic/cms#2967]
## Bug Description
I talked about this for a while with Duncan today on [Discord](https://discord.com/channels/489818810157891584/489819906540568593/784346071802970183). When I merge changes that the Git integration commited on the production branch to my develop repo I almost always get the error:

`Call to a member function id() on null`

The fix is usually a `php artisan cache:clear`, but this is pretty annoying as this usually happens multiple times a day. I feel it's bad for new comers as the error is vague and it doesn't present a solution. I can imagine this making v3 feel unstable.

I just triggered this error and I noticed the following stuff changed on the production branch:

- added pages to page structure
- removed pages from page structure
- new entries
- deleted entries
- new assets
- new form entries
- modified entries (content)
- modified entries (title)

## How to Reproduce
1. Modify stuff on production (I don't know what event triggers it though)
2. Let the git integration commit it.
3. Merge into your local dev branch.

## Environment
Statamic 3.0.32 Pro
Laravel 8.17.0
PHP 7.4.10
edalzell/forma 0.7.2
jonassiewertsen/statamic-livewire 2.2.0
silentz/mailchimp 2.4.1

**Install method (choose one):**
- Starter kit

**Comment by robdekort** [comment]
I could test different events some time to see what triggers it and what not. Let me know if that's welcome.

**Comment by robdekort** [comment]
Agreed. Don't think the git integration itself is the issue. But the branch switching/merging def. craps on the stache somehow.

**Comment by jasonvarga** [comment]
Do you have `watcher` enabled in `config/statamic/stache.php`?

**Comment by robdekort** [comment]
It’s off on production and on locally.

**Comment by jelleroorda** [comment]
Same for me, also off on production and on locally.

**Comment by damcclean** [comment]
I've also came across this issue numerous times before and it's a real pain, especially for those where it causes issues in a production environment when they deploy.

**Comment by sandersjj** [comment]
I have this same issue, but for this site I am not using the `pro` version (yet). Getting this error message:

`[2020-12-27 18:56:09] production.ERROR: Call to a member function id() on null {"exception":"[object] (Error(code: 0): Call to a member function id() on null at /home/forge/test.lekst.nl/vendor/statamic/cms/src/Stache/Stores/Store.php:79`

**Comment by damcclean** [comment]
> I have this same issue, but for this site I am not using the `pro` version (yet). Getting this error message:
> 
> 
> 
> `[2020-12-27 18:56:09] production.ERROR: Call to a member function id() on null {"exception":"[object] (Error(code: 0): Call to a member function id() on null at /home/forge/test.lekst.nl/vendor/statamic/cms/src/Stache/Stores/Store.php:79`

Does clearing your cache fix the issue for you?

**Comment by sandersjj** [comment]
No it doesn't.

**Comment by robdekort** [comment]
My client is reporting 500 error's on production as well so I had to give them access to the cache utility. It's hard to get useful information from them on when it craps out, but if you're interested in this site as a test case. It might also help debug the performance issues without caching as mentioned here #2704.

**Comment by johncarter-** [comment]
Has anyone got a decent work around for this?

**Comment by SimonMauvaisGarcons** [comment]
Hi, I had a very similar issue. 
 
```
Call to a member function id() on null at
.../vendor/statamic/cms/src/Stache/Stores/Store.php:79)
```

Juste like  @robdekort mentionne, the simple cache command `php artisan cache:clear`, normally work in most usecase _Call to a member function id() on null._  

Basically, I had duplicate entries in one of the collections. Some entries where dated and not the other. I removed the undesires entries and clear cache again: `php artisan cache:clear`. 

<img width="422" alt="Capture d’écran, le 2021-02-08 à 09 04 29" src="https://user-images.githubusercontent.com/20709665/107233332-791ddd80-69f0-11eb-95b0-4bd373eaca22.png">


Here's how I fixed my issue. 

1. I started by commenting the content folder: `_content`. This should remove the error. 
2. Then, I uncomment the content folder and I commented sub folders. I Continued this pattern until I found the content that was causing the error. The image above show the duplicate entries. 

Hope that can help. 
____

Statamic 3.0.42 Pro
Laravel 8.26.1
PHP 7.4.12
No addons installed
